pennylane/devices/null_qubit.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+The null.qubit device is a no-op device for benchmarking PennyLane's auxiliary functionality outside direct circuit evaluations.
+"""
+from collections import defaultdict
+
+from pennylane.devices import DefaultQubit
+from .._version import __version__
+
+# pylint: disable=unused-argument
+class NullQubit(DefaultQubit):
+    """Null qubit device for PennyLane.
+
+    Args:
+        wires (int, Iterable[Number, str]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers (i.e., ``[-1, 0, 2]``)
+            or strings (``['ancilla', 'q1', 'q2']``). Default 1 if not specified.
```

Comment:
```suggestion
            or strings (``['auxiliary', 'q1', 'q2']``). Default 1 if not specified.
```

The word `ancilla` is never to be used in any part of the code-base. We should remove all uses of this in favour of `auxiliary`